crates/matrix-sdk-crypto/src/machine.rs

```diff
@@ -1841,6 +1844,45 @@ impl OlmMachine {
         Ok(exported)
     }
 
+    /// Export room keys matching a predicate, providing them as an async
+    /// `Stream`.
+    ///
+    /// * `predicate` - A closure that will be called for every known
+    /// `InboundGroupSession`, which represents a room key. If the closure
+    /// returns `true` the `InboundGroupSession` will be included in the export,
+    /// if the closure returns `false` it will not be included.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use std::pin::pin;
+    ///
+    /// use matrix_sdk_crypto::{olm::ExportedRoomKey, OlmMachine};
+    /// use ruma::{device_id, room_id, user_id};
+    /// use tokio_stream::StreamExt;
+    /// # async {
+    /// let alice = user_id!("@alice:example.org");
+    /// let machine = OlmMachine::new(&alice, device_id!("DEVICEID")).await;
+    /// let room_id = room_id!("!test:localhost");
+    /// let mut keys = pin!(machine
+    ///     .export_room_keys_stream(|s| s.room_id() == room_id)
+    ///     .await
+    ///     .unwrap());
+    /// while let Some(key) = keys.next().await {
+    ///     println!("{}", key.room_id);
+    /// }
+    /// # };
+    /// ```
+    pub async fn export_room_keys_stream(
```

Comment:
Can we move this into `OlmMachine::store()` I'm trying to scope methods on the `OlmMachine` to some submodules so we don't have as many on the `OlmMachine` anymore.

Reply:
Yes. Will rebase on top of https://github.com/matrix-org/matrix-rust-sdk/pull/3146